File: LeetCode/Array/1200. Minimum Absolute Difference/Ingyu.java
import java.util.Arrays;
import java.util.LinkedList;
import java.util.List;

class Solution {
    public List<List<Integer>> minimumAbsDifference(int[] arr) {
        List<List<Integer>> answer = new LinkedList();
        Arrays.sort(arr);
        int pivot = Integer.MAX_VALUE;
        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i+1] - arr[i] < pivot) pivot = arr[i] - arr[i+1];
        }

        for (int i = 0; i < arr.length - 1; i++) {
            if (arr[i+1] - arr[i] == pivot) {
                List temp = new LinkedList();
                temp.add(arr[i]);
                temp.add(arr[i+1]);
                answer.add(temp);
            }
        }

        return answer;
    }
}

File: LeetCode/Array/1200. Minimum Absolute Difference/Ingyu.kt
import java.util.*

class Solution {
    fun minimumAbsDifference(arr: IntArray): List<List<Int>> {
        val answer = LinkedList<List<Int>>()
        arr.sort()

        var pivot = Integer.MAX_VALUE
        for (i in 0..arr.size-2) {
            if (arr[i+1] - arr[i] < pivot) pivot = arr[i+1] - arr[i]
        }

        for (i in 0..arr.size-2) {
            if (arr[i+1] - arr[i] == pivot) {
                val list = LinkedList<Int>()
                list.add(arr[i])
                list.add(arr[i+1])
                answer.add(list)
            }
        }

        return answer
    }
}